### PR TITLE
 feat(docker): Dockerfile and README update for basic docker development workflow

### PR DESCRIPTION
**Pull request by jamonation**
I've added a dockerfile and some instructions to README.md explaining how to use Docker to develop and run the oauth server. It allows for development with an IDE underneath a running container as one would normally work. This means changes and running the verifier are mostly transparent to normal development. If there are any questions feel free to ask here or in IRC.

### CODE REVIEW

**Comment by vladikoff** [comment]
@jamonation this needs to be rebased against `master`

**Comment by jamonation** [comment]
Thanks @vladikoff , I rebased that backwards. Should be all set now.

File: README.md [hunk 1]
```diff
@@ -14,3 +14,12 @@ Implementation of OAuth for use by Firefox Accounts
 
 MPL 2.0
 
+## Docker Based Development
+
+To run the profile server via Docker, two steps are required:
```
Comment (rfk) [inline README.md]: s/profile/oauth

File: README.md [hunk 2]
```diff
@@ -14,3 +14,12 @@ Implementation of OAuth for use by Firefox Accounts
...
+
+    $ docker build --rm -t mozilla/fxa_profile_server
```
Comment (rfk) [inline README.md]: also s/profile/oauth/ here

**Comment by rfk** [comment]
this looks pretty sweet, thanks for the contribution

**Pull request merged**